### PR TITLE
refactor: Improve `getProfileMetadata(...)`

### PR DESCRIPTION
**Pull request by b00ste**
# What does this PR introduce?

This PR improves the util `getProfileMetadata(...)` by introducing axios instead of fetch as it's more flexible with node version. Overload the util function so it is more flexible for end user. Add tests for the util function.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -41,6 +41,7 @@
     "dependencies": {
         "@erc725/erc725.js": "^0.20.0",
         "@lukso/lsp-smart-contracts": "^0.11.1",
+        "axios": "^1.5.1",
```
Comment (richtera) [inline package.json]: Unless you really need to use axios fetch might be a more standard api.

File: src/LSP3/getProfileMetadata.ts [hunk 2]
```diff
@@ -50,13 +69,15 @@ export const getProfileMetadata = async (unviersalProfile: UniversalProfile) =>
 
     let profileData: LSP3ProfileMetadata;
 
-    await fetch(profileDataURL)
-        .then(async (result) => await result.json())
-        .then((result) => (profileData = result));
+    await axios({
```
Comment (richtera) [inline src/LSP3/getProfileMetadata.ts]: Oh you're converting from fetch to axios. Note that workers only have fetch.

**Review by richtera — APPROVED** [review]
This looks ok, but I am curious why moving from fetch to axios. Fetch is available in browser and web workers.

File: src/LSP3/getProfileMetadata.ts (hunk 2 above)
Comment (b00ste) [reply src/LSP3/getProfileMetadata.ts]: I had some issues where fetch was missing in node 16 but present in node 18. But after updating to node 18 I managed to build but had some issues with tests, maybe we can double check and see if axios is really needed here.

**Comment by richtera** [comment]
Ah import it from isomorphic-fetch (that will automatically work node/browser)

**Review by richtera — APPROVED** [review]

**Pull request merged**